# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: acfafba3fdcc
# Full Hash: acfafba3fdcc02c781f1d3e04f219478a38a4912
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-03-13 04:24:24
# Regressor Bug: 1546759
# File Overlap Count: 1
# Description:
#   Bug 1546759 - Do not update session history entries in BC when it is discarded, r=peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66519
# ==============================================================================

diff -r 286dc40ec81b -r acfafba3fdcc dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Thu Mar 12 18:12:33 2020 +0000
+++ b/dom/ipc/ContentParent.cpp	Thu Mar 12 14:14:40 2020 +0000
@@ -5842,14 +5842,13 @@
 mozilla::ipc::IPCResult ContentParent::RecvUpdateSHEntriesInBC(
     PSHEntryParent* aNewLSHE, PSHEntryParent* aNewOSHE,
     const MaybeDiscarded<BrowsingContext>& aMaybeContext) {
-  if (aMaybeContext.IsNull()) {
+  if (aMaybeContext.IsNullOrDiscarded()) {
     MOZ_LOG(BrowsingContext::GetLog(), LogLevel::Debug,
             ("ParentIPC: Trying to update a browsing context that does not "
              "exist or has been discarded"));
     return IPC_OK();
   }
-  // if aContext has been discarded we can still update the entries
-  auto aContext = aMaybeContext.GetMaybeDiscarded()->Canonical();
+  auto aContext = aMaybeContext.get()->Canonical();
   MOZ_ASSERT(aContext);
   if (!aContext->IsOwnedByProcess(ChildID())) {
     // We are trying to update a child BrowsingContext in another child
